app/src/main/kotlin/com/wire/android/ui/common/UserProfileAvatar.kt

```diff
@@ -27,38 +27,48 @@ fun UserProfileAvatar(
     avatarUrl: String = "",
     size: Dp = MaterialTheme.wireDimensions.userAvatarDefaultSize,
     modifier: Modifier = Modifier,
+    isClickable: Boolean = true,
     onClick: () -> Unit = {}
 ) {
-    ConstraintLayout(
+    Box(
+        contentAlignment = Alignment.Center,
         modifier = modifier
-            .padding(10.dp)
-            .size(size)
+            .wrapContentSize()
+            .clip(CircleShape)
+            .then(if (isClickable) Modifier.clickable { onClick() } else Modifier)
+
     ) {
-        val (avatarImg, statusImg) = createRefs()
-        Image(
-            painter = painterResource(getAvatarAsDrawable(avatarUrl)),
-            contentDescription = stringResource(R.string.content_description_user_avatar),
-            contentScale = ContentScale.Crop,
-            modifier = Modifier
-                .background(Color.Black, CircleShape)
-                .clickable { onClick.invoke() }
-                .constrainAs(avatarImg) {
-                    top.linkTo(parent.top)
-                    bottom.linkTo(parent.bottom)
-                    start.linkTo(parent.start)
-                    end.linkTo(parent.end)
-                }
-        )
         Box(
-            modifier = Modifier
-                .size(10.dp)
-                .clip(CircleShape)
-                .background(Color.Green) // TODO: Map UserStatus availability to the right icon shape/color once this logic exists in Kalium
-                .constrainAs(statusImg) {
-                    bottom.linkTo(avatarImg.bottom)
-                    end.linkTo(avatarImg.end)
-                }
-        )
+            modifier = modifier
+                .wrapContentSize()
+                .padding(MaterialTheme.wireDimensions.userAvatarClickablePadding)
```

Comment:
Can't it be like this 
```
   Box(
        contentAlignment = Alignment.Center,
        modifier = modifier
            .wrapContentSize()
            .clip(CircleShape)
            .then(if (isClickable) Modifier.clickable { onClick() } else Modifier)
            .wrapContentSize()
            .padding(MaterialTheme.wireDimensions.userAvatarClickablePadding)
        ) { ....
```

instead of Box inside of the Box

Reply:
hmmm going to try

Reply:
yep, it works O.o